src/components/Registration/Registration.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { NavLink, useNavigate } from 'react-router-dom';
+
+import courseApi from '../../webClient';
+
+import '../../common/css/Common.css';
+import { TInputField } from 'src/types/inputField';
+import Input from 'src/common/Input/Input';
+
+export type TUserRequest = {
+	name: string;
+	email: string;
+	password: string;
+};
+
+const Registration = () => {
+	const navigate = useNavigate();
+	const [data, setData] = useState<TUserRequest>({
+		name: '',
+		email: '',
+		password: '',
+	});
+
+	const handleUpdate = (event) => {
+		const name = event.target.name;
+		const value = event.target.value;
+		setData((values) => ({ ...values, [name]: value }));
+	};
+
+	const createUser = (event) => {
+		event.preventDefault();
+		courseApi
+			.post('/register', JSON.stringify(data))
+			.then(function (response) {
+				navigate('/login');
+			})
+			.catch(function (error) {
+				// TODO add hanfler
+			});
+	};
+
+	const nameFieldSetup: TInputField = {
+		name: 'name',
+		label: 'Name',
+		type: 'text',
+		placeHolder: '',
+	};
+	const emailFieldSetup: TInputField = {
+		name: 'email',
+		label: 'Email',
+		type: 'email',
+		placeHolder: '',
+	};
+	const passwordFieldSetup: TInputField = {
+		name: 'password',
+		label: 'Password',
+		type: 'password',
+		placeHolder: '',
+	};
```

Comment:
Please collect it in an array and use below with map method and move it outside component body